zipline/sources/data_frame_source.py

```diff
@@ -108,11 +119,21 @@ class DataPanelSource(DataSource):
     def __init__(self, data, **kwargs):
         assert isinstance(data.major_axis, pd.tseries.index.DatetimeIndex)
 
-        self.data = data
+        self.data = data.fillna(method='ffill', axis=0)
```

Comment:
Re-wrote it to fill volume with 0's and ffill everything else. Does that make more sense?